In [1]:
!pip install pandas

Defaulting to user installation because normal site-packages is not writeable


In [2]:
import pandas as pd

In [3]:
# importing Sungrow file and removing the starting rows
df1 = pd.read_csv("10.csv",skiprows = 4)
df1.head(10)

Plant  Installed Power (kWp)  \
0                               Nitte                 483.00   
1              IBM Yeshwanthpur 52kwp                  52.00   
2  District Jail Tumakuru - 60.06 kWp                  60.06   
3     District Court Tumakuru - 86kWp                  70.00   
4      Gateway office block A3 140KWp                 140.00   
5       Decathlon kalamassery 80.1kwp                  70.00   
6           Decathlon Kottayam 40.5kw                  40.50   
7           Decathlon Guwahati 64.8kw                  64.80   
8           Decathlon _Vytila - 42kwp                  42.00   
9         Decathlon Hosur Road  70kwp                  70.00   

  Grid-connection Date  Yield Today (kWh)  Total Yield (kWh) Today Revenue  \
0           2023-10-25             1808.0           603366.9  1808.000 INR   
1           2023-07-13              132.0           256332.0   132.000 CNY   
2           2022-11-03              271.0            97499.8  1192.400 INR   
3           2022-11-03              363.9           129703.5  1965.060 INR   
4           2022-09-14              361.7           260534.5  2893.600 INR   
5           2022-03-31              247.8           313373.4  1239.000 INR   
6           2022-03-25              127.0           163485.7   571.500 INR   
7           2022-03-21               91.6           117783.7   412.200 INR   
8           2022-02-15              136.8           104705.5   136.800 CNY   
9           2022-03-15              247.9           167146.4   247.900 CNY   

   Total CO₂ reduction (kg)                                      Plant Address  
0                120871.993                                   NITTE University  
1                 25002.766                             Indian Bureau of Mines  
2                 94893.762  93PW+Q26, Service Rd, Antharasanahalli, Tumaku...  
3                121542.076  84W3+R4F, Ward No. 18, Tumkur, Tumakuru, Karna...  
4                220570.996                        16, Grand Southern Trunk Rd  
5                128474.118                        Decathlon Kalamassery Kochi  
6                 73788.668                                                  -  
7                117424.566             Ganakpara, Azara, Kamrup, Assam, India  
8                102180.337                            Decathlon Vyttila Kochi  
9                152534.220                               Decathlon Hosur Road

In [4]:
# checking the columns header names
df1.columns

Index(['Plant', 'Installed Power (kWp)', 'Grid-connection Date',
       'Yield Today (kWh)', 'Total Yield (kWh)', 'Today Revenue',
       'Total CO₂ reduction (kg)', 'Plant Address'],
      dtype='object')

In [5]:
#removing the extra columns
df1= df1[['Plant', 'Installed Power (kWp)', 'Yield Today (kWh)']]
# checking the table
df1.head(10)

Plant  Installed Power (kWp)  \
0                               Nitte                 483.00   
1              IBM Yeshwanthpur 52kwp                  52.00   
2  District Jail Tumakuru - 60.06 kWp                  60.06   
3     District Court Tumakuru - 86kWp                  70.00   
4      Gateway office block A3 140KWp                 140.00   
5       Decathlon kalamassery 80.1kwp                  70.00   
6           Decathlon Kottayam 40.5kw                  40.50   
7           Decathlon Guwahati 64.8kw                  64.80   
8           Decathlon _Vytila - 42kwp                  42.00   
9         Decathlon Hosur Road  70kwp                  70.00   

   Yield Today (kWh)  
0             1808.0  
1              132.0  
2              271.0  
3              363.9  
4              361.7  
5              247.8  
6              127.0  
7               91.6  
8              136.8  
9              247.9

In [6]:
# checking the columns header names
df1.columns

Index(['Plant', 'Installed Power (kWp)', 'Yield Today (kWh)'], dtype='object')

In [7]:
# Renaming required columns
df2 = df1.rename(columns={
    'Plant': 'Site Name',
    'Installed Power (kWp)': 'DC Capacity',
    'Yield Today (kWh)': 'Generation'
})
# checking the table
df2.head(10)

Site Name  DC Capacity  Generation
0                               Nitte       483.00      1808.0
1              IBM Yeshwanthpur 52kwp        52.00       132.0
2  District Jail Tumakuru - 60.06 kWp        60.06       271.0
3     District Court Tumakuru - 86kWp        70.00       363.9
4      Gateway office block A3 140KWp       140.00       361.7
5       Decathlon kalamassery 80.1kwp        70.00       247.8
6           Decathlon Kottayam 40.5kw        40.50       127.0
7           Decathlon Guwahati 64.8kw        64.80        91.6
8           Decathlon _Vytila - 42kwp        42.00       136.8
9         Decathlon Hosur Road  70kwp        70.00       247.9

In [8]:
# importing the wattmon file and removing the row 1
df51 = pd.read_csv("50.csv",skiprows = 1)
df51.head(10)

Site ID                 Time  kW_DC_Capacity  \
0                 FM  2024-01-01 00:00:00          171.14   
1              SJS 2  2024-01-01 00:00:00            0.00   
2              SJS 1  2024-01-01 00:00:00            0.00   
3                IBS  2024-01-01 00:00:00          120.00   
4     Flipkart_Hosur  2024-01-01 00:00:00          100.00   
5    Flipkart_Anekal  2024-01-01 00:00:00          150.00   
6            Acharya  2024-01-01 00:00:00          330.00   
7     SEL 14-CHENNAI  2024-01-01 00:00:00          490.00   
8               LIPL  2024-01-01 00:00:00           83.00   
9  Flipkart_Guwahati  2024-01-01 00:00:00           83.00   

   kWh_Daily_Generation  kWh_Day_Irradiance  
0               511.400               3.896  
1                 0.000               0.000  
2                 0.000               0.000  
3                 0.002               0.000  
4               365.000               4.934  
5               523.000               4.410  
6              1272.000               4.471  
7                 0.000               0.000  
8                 0.000               0.000  
9               271.000               3.125

In [9]:
df51.columns

Index(['Site ID', 'Time', 'kW_DC_Capacity', 'kWh_Daily_Generation',
       'kWh_Day_Irradiance'],
      dtype='object')

In [10]:
#removing the extra columns
df51= df51[['Site ID', 'kW_DC_Capacity', 'kWh_Daily_Generation','kWh_Day_Irradiance']]
# checking the table for above code
df51.head(10)

Site ID  kW_DC_Capacity  kWh_Daily_Generation  kWh_Day_Irradiance
0                 FM          171.14               511.400               3.896
1              SJS 2            0.00                 0.000               0.000
2              SJS 1            0.00                 0.000               0.000
3                IBS          120.00                 0.002               0.000
4     Flipkart_Hosur          100.00               365.000               4.934
5    Flipkart_Anekal          150.00               523.000               4.410
6            Acharya          330.00              1272.000               4.471
7     SEL 14-CHENNAI          490.00                 0.000               0.000
8               LIPL           83.00                 0.000               0.000
9  Flipkart_Guwahati           83.00               271.000               3.125

In [11]:
# Rename multiple columns
df52 = df51.rename(columns={
    'Site ID': 'Site Name',
    'kW_DC_Capacity': 'DC Capacity',
    'kWh_Daily_Generation': 'Generation',
    'kWh_Day_Irradiance': 'Irradiance'
})
df52.head(10)

Site Name  DC Capacity  Generation  Irradiance
0                 FM       171.14     511.400       3.896
1              SJS 2         0.00       0.000       0.000
2              SJS 1         0.00       0.000       0.000
3                IBS       120.00       0.002       0.000
4     Flipkart_Hosur       100.00     365.000       4.934
5    Flipkart_Anekal       150.00     523.000       4.410
6            Acharya       330.00    1272.000       4.471
7     SEL 14-CHENNAI       490.00       0.000       0.000
8               LIPL        83.00       0.000       0.000
9  Flipkart_Guwahati        83.00     271.000       3.125

In [12]:
# Appending the both files sungrow and wattmon
result = pd.concat([df2, df52], ignore_index=True)
df100=result
df100.head(55)

Site Name  DC Capacity  Generation  \
0                                              Nitte       483.00    1808.000   
1                             IBM Yeshwanthpur 52kwp        52.00     132.000   
2                 District Jail Tumakuru - 60.06 kWp        60.06     271.000   
3                    District Court Tumakuru - 86kWp        70.00     363.900   
4                     Gateway office block A3 140KWp       140.00     361.700   
5                      Decathlon kalamassery 80.1kwp        70.00     247.800   
6                          Decathlon Kottayam 40.5kw        40.50     127.000   
7                          Decathlon Guwahati 64.8kw        64.80      91.600   
8                          Decathlon _Vytila - 42kwp        42.00     136.800   
9                        Decathlon Hosur Road  70kwp        70.00     247.900   
10            Gateway office park A4 block - 121 KWp       121.00     340.500   
11                 Gateway office A2 Block - 140 kWp       140.00       0.000   
12                          futura tech park - 33kwp        33.00     138.400   
13                      Decathlon - Dehradun -140kwp       140.00     305.900   
14                           SRO-TSCL Tumkur (18.76)        18.76      67.900   
15                            ZPO-TSCL Tumkur(24.12)        24.12     100.600   
16                        DHFW-TSCL Tumkur(40.87kWp)        40.87     172.200   
17  GOKALDAS (ATLANTIC APPARELS) MYSURU [378.84] Kwp       378.84    1251.300   
18                                     CPDO(85.8kwp)        85.80     325.600   
19                        Big basket chakan (325Kwp)       325.00    1047.840   
20                            GSI BANGALORE (104.65)       104.65     340.796   
21                                                FM       171.14     511.400   
22                                             SJS 2         0.00       0.000   
23                                             SJS 1         0.00       0.000   
24                                               IBS       120.00       0.002   
25                                    Flipkart_Hosur       100.00     365.000   
26                                   Flipkart_Anekal       150.00     523.000   
27                                           Acharya       330.00    1272.000   
28                                    SEL 14-CHENNAI       490.00       0.000   
29                                              LIPL        83.00       0.000   
30                                 Flipkart_Guwahati        83.00     271.000   
31                                  SEL 15-Rajasthan         0.00       0.000   
32                                        ZF_CHENNAI       635.00       0.000   
33                                     ZF_COIMBATORE         0.00       0.000   
34                                 Flipkart_Anjaneya       320.00    1377.000   
35                                  Flipkart_Chennai       500.00     759.000   
36                                 Flipkart_Uluberia       270.00     459.000   
37                                  Flipkart_Kolkata       130.00     225.000   
38                          Flipkart_Saidham_Grocery       390.00     421.000   
39                            Flipkart_Saidham_Large       450.00    1259.000   
40                               Flipkart_Saidham_MH       650.00    1395.000   
41                               Flipkart_Saidham_FC       650.00    2107.000   
42                                  BigBasket_Mysuru       125.00     494.000   
43                                   Dmart_Gollapudi       151.00     426.000   
44                                      SEL 16-Kolar       297.48    1169.000   
45                                           Army_AB       116.00     229.000   
46                                           Army_LU        84.00     162.000   
47                                       Dmart_Vasai        83.00     284.000   
48                                       Dmart_Virar        81.00     295.

### Save the DataFrame to an Excel file
df100.to_excel("output.xlsx", index=False)

In [13]:
# importing the solaredge file and removing the starting rows
df21 = pd.read_excel("20.xlsx",skiprows = 9)
df21 = df21[:-1]
df21.head()

C:\ProgramData\anaconda3\Lib\site-packages\openpyxl\styles\stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


#                       Site Name  Peak Power (kWp)  Highest Alert Impact  \
0  1  Gokaldas Exports Ltd. - Mysore             21.12                   0.0   
1  2           Karnal - Globe Toyota            142.80                   2.0   
2  3      Yamunanagar - Globe Toyota            143.65                   1.0   

    kWh  kWh/kWp  
0  76.3      3.6  
1   0.0      0.0  
2   0.4      0.0

In [14]:
df21.columns

Index(['#', 'Site Name', 'Peak Power (kWp)', 'Highest Alert Impact', 'kWh',
       'kWh/kWp'],
      dtype='object')

In [15]:
#removing the extra columns
df21= df21[['Site Name','Peak Power (kWp)','kWh']]
df21.head()

Site Name  Peak Power (kWp)   kWh
0  Gokaldas Exports Ltd. - Mysore             21.12  76.3
1           Karnal - Globe Toyota            142.80   0.0
2      Yamunanagar - Globe Toyota            143.65   0.4

In [16]:
# Rename multiple columns
df22 = df21.rename(columns={
    'Site Name': 'Site Name',
    'Peak Power (kWp)': 'DC Capacity',
    'kWh': 'Generation'
})
df22.head(10)

Site Name  DC Capacity  Generation
0  Gokaldas Exports Ltd. - Mysore        21.12        76.3
1           Karnal - Globe Toyota       142.80         0.0
2      Yamunanagar - Globe Toyota       143.65         0.4

In [17]:
# Appending the 3 files sungrow,wattmon,solaredge
result1 = pd.concat([df2,df52,df22], ignore_index=True)
df200=result1
df200.head(60)

Site Name  DC Capacity  Generation  \
0                                              Nitte       483.00    1808.000   
1                             IBM Yeshwanthpur 52kwp        52.00     132.000   
2                 District Jail Tumakuru - 60.06 kWp        60.06     271.000   
3                    District Court Tumakuru - 86kWp        70.00     363.900   
4                     Gateway office block A3 140KWp       140.00     361.700   
5                      Decathlon kalamassery 80.1kwp        70.00     247.800   
6                          Decathlon Kottayam 40.5kw        40.50     127.000   
7                          Decathlon Guwahati 64.8kw        64.80      91.600   
8                          Decathlon _Vytila - 42kwp        42.00     136.800   
9                        Decathlon Hosur Road  70kwp        70.00     247.900   
10            Gateway office park A4 block - 121 KWp       121.00     340.500   
11                 Gateway office A2 Block - 140 kWp       140.00       0.000   
12                          futura tech park - 33kwp        33.00     138.400   
13                      Decathlon - Dehradun -140kwp       140.00     305.900   
14                           SRO-TSCL Tumkur (18.76)        18.76      67.900   
15                            ZPO-TSCL Tumkur(24.12)        24.12     100.600   
16                        DHFW-TSCL Tumkur(40.87kWp)        40.87     172.200   
17  GOKALDAS (ATLANTIC APPARELS) MYSURU [378.84] Kwp       378.84    1251.300   
18                                     CPDO(85.8kwp)        85.80     325.600   
19                        Big basket chakan (325Kwp)       325.00    1047.840   
20                            GSI BANGALORE (104.65)       104.65     340.796   
21                                                FM       171.14     511.400   
22                                             SJS 2         0.00       0.000   
23                                             SJS 1         0.00       0.000   
24                                               IBS       120.00       0.002   
25                                    Flipkart_Hosur       100.00     365.000   
26                                   Flipkart_Anekal       150.00     523.000   
27                                           Acharya       330.00    1272.000   
28                                    SEL 14-CHENNAI       490.00       0.000   
29                                              LIPL        83.00       0.000   
30                                 Flipkart_Guwahati        83.00     271.000   
31                                  SEL 15-Rajasthan         0.00       0.000   
32                                        ZF_CHENNAI       635.00       0.000   
33                                     ZF_COIMBATORE         0.00       0.000   
34                                 Flipkart_Anjaneya       320.00    1377.000   
35                                  Flipkart_Chennai       500.00     759.000   
36                                 Flipkart_Uluberia       270.00     459.000   
37                                  Flipkart_Kolkata       130.00     225.000   
38                          Flipkart_Saidham_Grocery       390.00     421.000   
39                            Flipkart_Saidham_Large       450.00    1259.000   
40                               Flipkart_Saidham_MH       650.00    1395.000   
41                               Flipkart_Saidham_FC       650.00    2107.000   
42                                  BigBasket_Mysuru       125.00     494.000   
43                                   Dmart_Gollapudi       151.00     426.000   
44                                      SEL 16-Kolar       297.48    1169.000   
45                                           Army_AB       116.00     229.000   
46                                           Army_LU        84.00     162.000   
47                                       Dmart_Vasai        83.00     284.000   
48                                       Dmart_Virar        81.00     295.

In [18]:
# importing the Huawei file and removing the starting rows
df31 = pd.read_excel("30.xlsx",skiprows = 1)
df31.head()

Plant Name  \
0  U Solar_Big Basket_Kundali 399 Kwp   
1               U Solar_Polyone-550KW   

                                       Address  Total String Capacity (kWp)  \
0                     Kundli, Sonepat, Haryana                      399.400   
1  Bhiwandi, Thane, Maharashtra, 421302, India                      550.405   

   Global Irradiation (kWh/㎡)  Sun Hours (H)  Average Temperature(°C)  \
0                       0.934          6.333                      0.0   
1                         NaN            NaN                      NaN   

   Theoretical Yield (kWh)  PV Yield (kWh)   Inverter Yield (kWh)  \
0                  372.884          396.85                 396.85   
1                      NaN         1606.22                1606.22   

   Total Yield (kWh)  ...  Loss Due to Export Limitation(Rs.)  \
0          936867.52  ...                                 NaN   
1          656633.81  ...                                 NaN   

   Consumption (kWh)  Self-consumption (kWh)  Self-consumption Rate (%)  \
0             396.85                  396.85                        NaN   
1            1606.22                     NaN                        NaN   

   Peak Power (kW)  Performance Ratio(%)  Load Rate (%)  CO₂ Avoided (t)  \
0           95.064                 100.0         23.802            0.189   
1          261.530                   NaN         47.516            0.763   

   Standard Coal Saved (t)  Revenue (Rs.)  
0                    0.159         3174.8  
1                    0.642            NaN  

[2 rows x 24 columns]

In [19]:
df31.columns

Index(['Plant Name', 'Address', 'Total String Capacity (kWp)',
       'Global Irradiation (kWh/㎡)', 'Sun Hours (H)',
       'Average Temperature(°C)', 'Theoretical Yield (kWh)', 'PV Yield (kWh)',
       ' Inverter Yield (kWh)', 'Total Yield (kWh)', 'Export (kWh)',
       'Specific Energy (kWh/kWp)', 'Import (kWh)',
       'Loss Due to Export Limitation (kWh)',
       'Loss Due to Export Limitation(Rs.)', 'Consumption (kWh)',
       'Self-consumption (kWh)', 'Self-consumption Rate (%)',
       'Peak Power (kW)', 'Performance Ratio(%)', 'Load Rate (%)',
       'CO₂ Avoided (t)', 'Standard Coal Saved (t)', 'Revenue (Rs.)'],
      dtype='object')

In [20]:
#removing the extra columns
df31= df31[['Plant Name','Total String Capacity (kWp)','PV Yield (kWh)']]
df31.head()

Plant Name  Total String Capacity (kWp)  \
0  U Solar_Big Basket_Kundali 399 Kwp                      399.400   
1               U Solar_Polyone-550KW                      550.405   

   PV Yield (kWh)  
0          396.85  
1         1606.22

In [21]:
# Rename multiple columns
df32 = df31.rename(columns={
    'Plant Name': 'Site Name',
    'Total String Capacity (kWp)': 'DC Capacity',
    'PV Yield (kWh)': 'Generation'
})
df32.head(10)

Site Name  DC Capacity  Generation
0  U Solar_Big Basket_Kundali 399 Kwp      399.400      396.85
1               U Solar_Polyone-550KW      550.405     1606.22

In [22]:
# Appending the 4 files sungrow,wattmon,solaredge,huawei
result2 = pd.concat([df2,df22,df32,df52], ignore_index=True)
df300=result2
df300.head(70)

Site Name  DC Capacity  Generation  Irradiance
0                                Nitte       483.00      1808.0         NaN
1               IBM Yeshwanthpur 52kwp        52.00       132.0         NaN
2   District Jail Tumakuru - 60.06 kWp        60.06       271.0         NaN
3      District Court Tumakuru - 86kWp        70.00       363.9         NaN
4       Gateway office block A3 140KWp       140.00       361.7         NaN
..                                 ...          ...         ...         ...
56                Bigbasket_Kolkata_01       134.00       372.0       0.000
57                Bigbasket_Kolkata_02       116.00       326.0       3.138
58                     Dmart_Perungudi         0.00         0.0       0.000
59                    SEL 11-Ahmedabad       420.00       963.0       0.000
60                      GT Yamunanagar         0.00         0.0       0.000

[61 rows x 4 columns]

# Intermediate Output-1

# Save the DataFrame to an Excel file
df300.to_excel("output2.xlsx", index=True, startrow=1)

In [23]:
df300.columns

Index(['Site Name', 'DC Capacity', 'Generation', 'Irradiance'], dtype='object')

In [24]:
# Adding Yield column in DataFrame
df300['Yield'] = df300['Generation'] / df300['DC Capacity']
# Round the 'Yield' column to 2 decimal places
df300['Yield'] = df300['Yield'].round(2)
df300.head()


Site Name  DC Capacity  Generation  Irradiance  \
0                               Nitte       483.00      1808.0         NaN   
1              IBM Yeshwanthpur 52kwp        52.00       132.0         NaN   
2  District Jail Tumakuru - 60.06 kWp        60.06       271.0         NaN   
3     District Court Tumakuru - 86kWp        70.00       363.9         NaN   
4      Gateway office block A3 140KWp       140.00       361.7         NaN   

   Yield  
0   3.74  
1   2.54  
2   4.51  
3   5.20  
4   2.58

In [25]:
# adding PR column to df300 
df300['PR'] = (df300['Yield'] / df300['Irradiance'])*100
# Round the 'PR' column to 2 decimal places
df300['PR'] = df300['PR'].round(2)
df300.head()

Site Name  DC Capacity  Generation  Irradiance  \
0                               Nitte       483.00      1808.0         NaN   
1              IBM Yeshwanthpur 52kwp        52.00       132.0         NaN   
2  District Jail Tumakuru - 60.06 kWp        60.06       271.0         NaN   
3     District Court Tumakuru - 86kWp        70.00       363.9         NaN   
4      Gateway office block A3 140KWp       140.00       361.7         NaN   

   Yield  PR  
0   3.74 NaN  
1   2.54 NaN  
2   4.51 NaN  
3   5.20 NaN  
4   2.58 NaN

In [26]:
# Replace NaN or empty cells with 0
df300 = df300.fillna(0)

# Replace infinity (inf) values with 0 in the 'PR' column
df300['PR'] = df300['PR'].replace([float('inf')], 0)
df300.head()

Site Name  DC Capacity  Generation  Irradiance  \
0                               Nitte       483.00      1808.0         0.0   
1              IBM Yeshwanthpur 52kwp        52.00       132.0         0.0   
2  District Jail Tumakuru - 60.06 kWp        60.06       271.0         0.0   
3     District Court Tumakuru - 86kWp        70.00       363.9         0.0   
4      Gateway office block A3 140KWp       140.00       361.7         0.0   

   Yield   PR  
0   3.74  0.0  
1   2.54  0.0  
2   4.51  0.0  
3   5.20  0.0  
4   2.58  0.0

In [27]:
# Wattmon capex sites removing
values_to_remove = ['FM', 'SJS 2', 'SJS 1', 'IBS','SEL 14-CHENNAI','LIPL','SEL 15-Rajasthan','ZF_CHENNAI','ZF_COIMBATORE','SEL 16-Kolar','SEL 8-Bommasandra','SEL 11-Ahmedabad','GT Yamunanagar']

# Remove rows with the specified values in the 'Site Name' column
df300 = df300[~df300['Site Name'].isin(values_to_remove)]
df300.head(60)

Site Name  DC Capacity  Generation  \
0                                              Nitte      483.000    1808.000   
1                             IBM Yeshwanthpur 52kwp       52.000     132.000   
2                 District Jail Tumakuru - 60.06 kWp       60.060     271.000   
3                    District Court Tumakuru - 86kWp       70.000     363.900   
4                     Gateway office block A3 140KWp      140.000     361.700   
5                      Decathlon kalamassery 80.1kwp       70.000     247.800   
6                          Decathlon Kottayam 40.5kw       40.500     127.000   
7                          Decathlon Guwahati 64.8kw       64.800      91.600   
8                          Decathlon _Vytila - 42kwp       42.000     136.800   
9                        Decathlon Hosur Road  70kwp       70.000     247.900   
10            Gateway office park A4 block - 121 KWp      121.000     340.500   
11                 Gateway office A2 Block - 140 kWp      140.000       0.000   
12                          futura tech park - 33kwp       33.000     138.400   
13                      Decathlon - Dehradun -140kwp      140.000     305.900   
14                           SRO-TSCL Tumkur (18.76)       18.760      67.900   
15                            ZPO-TSCL Tumkur(24.12)       24.120     100.600   
16                        DHFW-TSCL Tumkur(40.87kWp)       40.870     172.200   
17  GOKALDAS (ATLANTIC APPARELS) MYSURU [378.84] Kwp      378.840    1251.300   
18                                     CPDO(85.8kwp)       85.800     325.600   
19                        Big basket chakan (325Kwp)      325.000    1047.840   
20                            GSI BANGALORE (104.65)      104.650     340.796   
21                    Gokaldas Exports Ltd. - Mysore       21.120      76.300   
22                             Karnal - Globe Toyota      142.800       0.000   
23                        Yamunanagar - Globe Toyota      143.650       0.400   
24                U Solar_Big Basket_Kundali 399 Kwp      399.400     396.850   
25                             U Solar_Polyone-550KW      550.405    1606.220   
30                                    Flipkart_Hosur      100.000     365.000   
31                                   Flipkart_Anekal      150.000     523.000   
32                                           Acharya      330.000    1272.000   
35                                 Flipkart_Guwahati       83.000     271.000   
39                                 Flipkart_Anjaneya      320.000    1377.000   
40                                  Flipkart_Chennai      500.000     759.000   
41                                 Flipkart_Uluberia      270.000     459.000   
42                                  Flipkart_Kolkata      130.000     225.000   
43                          Flipkart_Saidham_Grocery      390.000     421.000   
44                            Flipkart_Saidham_Large      450.000    1259.000   
45                               Flipkart_Saidham_MH      650.000    1395.000   
46                               Flipkart_Saidham_FC      650.000    2107.000   
47                                  BigBasket_Mysuru      125.000     494.000   
48                                   Dmart_Gollapudi      151.000     426.000   
50                                           Army_AB      116.000     229.000   
51                                           Army_LU       84.000     162.000   
52                                       Dmart_Vasai       83.000     284.000   
53                                       Dmart_Virar       81.000     295.000   
54                                  Dmart_Pandharpur      198.000     801.000   
56                              Bigbasket_Kolkata_01      134.000     372.000   
57                              Bigbasket_Kolkata_02      116.000     326.000   
58                                   Dmart_Perungudi        0.000       0.000   

    Irradiance  Yield      PR  
0        0.000   3.74    0.00  
1        

In [28]:
# Sort the DataFrame in descending order by 'DC Capacity' column
df300 = df300.sort_values(by='DC Capacity', ascending=False)

# Display the sorted DataFrame
df300


Site Name  DC Capacity  Generation  \
46                               Flipkart_Saidham_FC      650.000    2107.000   
45                               Flipkart_Saidham_MH      650.000    1395.000   
25                             U Solar_Polyone-550KW      550.405    1606.220   
40                                  Flipkart_Chennai      500.000     759.000   
0                                              Nitte      483.000    1808.000   
44                            Flipkart_Saidham_Large      450.000    1259.000   
24                U Solar_Big Basket_Kundali 399 Kwp      399.400     396.850   
43                          Flipkart_Saidham_Grocery      390.000     421.000   
17  GOKALDAS (ATLANTIC APPARELS) MYSURU [378.84] Kwp      378.840    1251.300   
32                                           Acharya      330.000    1272.000   
19                        Big basket chakan (325Kwp)      325.000    1047.840   
39                                 Flipkart_Anjaneya      320.000    1377.000   
41                                 Flipkart_Uluberia      270.000     459.000   
54                                  Dmart_Pandharpur      198.000     801.000   
48                                   Dmart_Gollapudi      151.000     426.000   
31                                   Flipkart_Anekal      150.000     523.000   
23                        Yamunanagar - Globe Toyota      143.650       0.400   
22                             Karnal - Globe Toyota      142.800       0.000   
13                      Decathlon - Dehradun -140kwp      140.000     305.900   
11                 Gateway office A2 Block - 140 kWp      140.000       0.000   
4                     Gateway office block A3 140KWp      140.000     361.700   
56                              Bigbasket_Kolkata_01      134.000     372.000   
42                                  Flipkart_Kolkata      130.000     225.000   
47                                  BigBasket_Mysuru      125.000     494.000   
10            Gateway office park A4 block - 121 KWp      121.000     340.500   
50                                           Army_AB      116.000     229.000   
57                              Bigbasket_Kolkata_02      116.000     326.000   
20                            GSI BANGALORE (104.65)      104.650     340.796   
30                                    Flipkart_Hosur      100.000     365.000   
18                                     CPDO(85.8kwp)       85.800     325.600   
51                                           Army_LU       84.000     162.000   
35                                 Flipkart_Guwahati       83.000     271.000   
52                                       Dmart_Vasai       83.000     284.000   
53                                       Dmart_Virar       81.000     295.000   
5                      Decathlon kalamassery 80.1kwp       70.000     247.800   
9                        Decathlon Hosur Road  70kwp       70.000     247.900   
3                    District Court Tumakuru - 86kWp       70.000     363.900   
7                          Decathlon Guwahati 64.8kw       64.800      91.600   
2                 District Jail Tumakuru - 60.06 kWp       60.060     271.000   
1                             IBM Yeshwanthpur 52kwp       52.000     132.000   
8                          Decathlon _Vytila - 42kwp       42.000     136.800   
16                        DHFW-TSCL Tumkur(40.87kWp)       40.870     172.200   
6                          Decathlon Kottayam 40.5kw       40.500     127.000   
12                          futura tech park - 33kwp       33.000     138.400   
15                            ZPO-TSCL Tumkur(24.12)       24.120     100.600   
21                    Gokaldas Exports Ltd. - Mysore       21.120      76.300   
14                           SRO-TSCL Tumkur (18.76)       18.760      67.900   
58                                   Dmart_Perungudi        0.000       0.000   

    Irradiance  Yield      PR  
46       7.601   3.24   42.63  
45       

In [29]:
# Reset the index and remove the old index column
df400 = df300.reset_index(drop=True)

# Create the 'Serial No' column
df400.insert(0, 'Serial No', range(1, len(df300) + 1))

df400

Serial No                                         Site Name  DC Capacity  \
0           1                               Flipkart_Saidham_FC      650.000   
1           2                               Flipkart_Saidham_MH      650.000   
2           3                             U Solar_Polyone-550KW      550.405   
3           4                                  Flipkart_Chennai      500.000   
4           5                                             Nitte      483.000   
5           6                            Flipkart_Saidham_Large      450.000   
6           7                U Solar_Big Basket_Kundali 399 Kwp      399.400   
7           8                          Flipkart_Saidham_Grocery      390.000   
8           9  GOKALDAS (ATLANTIC APPARELS) MYSURU [378.84] Kwp      378.840   
9          10                                           Acharya      330.000   
10         11                        Big basket chakan (325Kwp)      325.000   
11         12                                 Flipkart_Anjaneya      320.000   
12         13                                 Flipkart_Uluberia      270.000   
13         14                                  Dmart_Pandharpur      198.000   
14         15                                   Dmart_Gollapudi      151.000   
15         16                                   Flipkart_Anekal      150.000   
16         17                        Yamunanagar - Globe Toyota      143.650   
17         18                             Karnal - Globe Toyota      142.800   
18         19                      Decathlon - Dehradun -140kwp      140.000   
19         20                 Gateway office A2 Block - 140 kWp      140.000   
20         21                    Gateway office block A3 140KWp      140.000   
21         22                              Bigbasket_Kolkata_01      134.000   
22         23                                  Flipkart_Kolkata      130.000   
23         24                                  BigBasket_Mysuru      125.000   
24         25            Gateway office park A4 block - 121 KWp      121.000   
25         26                                           Army_AB      116.000   
26         27                              Bigbasket_Kolkata_02      116.000   
27         28                            GSI BANGALORE (104.65)      104.650   
28         29                                    Flipkart_Hosur      100.000   
29         30                                     CPDO(85.8kwp)       85.800   
30         31                                           Army_LU       84.000   
31         32                                 Flipkart_Guwahati       83.000   
32         33                                       Dmart_Vasai       83.000   
33         34                                       Dmart_Virar       81.000   
34         35                     Decathlon kalamassery 80.1kwp       70.000   
35         36                       Decathlon Hosur Road  70kwp       70.000   
36         37                   District Court Tumakuru - 86kWp       70.000   
37         38                         Decathlon Guwahati 64.8kw       64.800   
38         39                District Jail Tumakuru - 60.06 kWp       60.060   
39         40                            IBM Yeshwanthpur 52kwp       52.000   
40         41                         Decathlon _Vytila - 42kwp       42.000   
41         42                        DHFW-TSCL Tumkur(40.87kWp)       40.870   
42         43                         Decathlon Kottayam 40.5kw       40.500   
43         44                          futura tech park - 33kwp       33.000   
44         45                            ZPO-TSCL Tumkur(24.12)       24.120   
45         46                    Gokaldas Exports Ltd. - Mysore       21.120   
46         47                           SRO-TSCL Tumkur (18.76)       18.760   
47         48                                   Dmart_Perungudi        0.000   

    Generation  Irradiance  Yield      PR  
0     2107.000       7.601   3.24   42.

#removing the extra columns
df300= df300[[ 'Serial No','Site Name', 'DC Capacity', 'Generation', 'Irradiance', 'Yield', 'PR'
      ]]
df300.head()

# Save the DataFrame to an Excel file
df300.to_excel("output10.xlsx", index=True, startrow=1)

### Sort the DataFrame in descending order by 'DC Capacity'
df400 = df300.sort_values(by='DC Capacity', ascending=False)

# Reset the index and rename the index column
df400 = df400.reset_index(drop=True)
df400.index = df400.index + 1
df400.index.name = 'S No'
df400.head()

In [30]:
df400.columns

Index(['Serial No', 'Site Name', 'DC Capacity', 'Generation', 'Irradiance',
       'Yield', 'PR'],
      dtype='object')

In [31]:
# Save the DataFrame to an Excel file
#df300.to_excel("DGR.xlsx", index=False, startrow=1)

In [32]:
# Remove rows with non-numeric values in the columns you want to sum
df = df400[pd.to_numeric(df400['DC Capacity'], errors='coerce').notna()]
df = df400[pd.to_numeric(df400['Generation'], errors='coerce').notna()]

# Calculate the sum for 'DC Capacity' and 'Generation' columns
sum_values = df[['DC Capacity', 'Generation']].sum()

# Calculate the average for 'Irradiance', 'Yield', and 'PR' columns
average_values = df[['Irradiance', 'Yield', 'PR']].mean()

# Add a new row with 'Total' values using the loc method
df.loc['Total'] = ['','Total', sum_values['DC Capacity'], sum_values['Generation'],
                   average_values['Irradiance'], average_values['Yield'], average_values['PR']]

# Save the updated DataFrame to a new Excel file
#df.to_excel("DGR_with_summary.xlsx", index=True)


In [33]:
# Convert the 'Yield' column to numeric (if it contains numeric values)
df400['Yield'] = pd.to_numeric(df400['Yield'], errors='coerce')

In [34]:
# Define a function to apply cell colors based on values
def highlight_yield(value):
    if 0 <= value < 3:
        return 'background-color: lightcoral'
    elif 3 <= value < 4:
        return 'background-color: #fcf75e'
    elif 4 <= value < 6:
        return 'background-color: lightgreen'
    else:
        return ''

# Apply the cell coloring to the 'Yield' column
styled_df = df.style.applymap(highlight_yield, subset=['Yield'])

# Display the styled DataFrame
styled_df



# Define a function to apply cell colors based on values in the 'PR' column
def highlight_pr(value):
    if 0 <= value < 60:
        return 'background-color: lightcoral'
    elif 60 <= value < 80:
        return 'background-color: palegoldenrod'
    elif 80 <= value < 150:
        return 'background-color: lightgreen'
    else:
        return ''

# Apply the cell coloring to the 'PR' column
styled_df = styled_df.applymap(highlight_pr, subset=['PR'])

# Display the styled DataFrame
styled_df


In [35]:
# Save the DataFrame to an Excel file
styled_df.to_excel("DGR.xlsx", index=False, startrow=1)


# Sort the DataFrame in descending order by 'DC Capacity' column
df = df.sort_values(by='DC Capacity', ascending=False)

# Display the sorted DataFrame
df


# Save the sorted DataFrame to an Excel file without the index column
df.to_excel("sorted_DGR.xlsx", index=False)
